cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelperTest.java

```diff
@@ -206,6 +215,81 @@ public void testAddingThrottlesWithPreExistingThrottles() throws InterruptedExce
+        new ReplicaPlacementInfo(0),
+        Arrays.asList(new ReplicaPlacementInfo(0), new ReplicaPlacementInfo(3)),
+        Arrays.asList(new ReplicaPlacementInfo(0), new ReplicaPlacementInfo(2)));
+
+    throttleHelper.setThrottles(Arrays.asList(proposal, proposal2));
+
+    ExecutionTask completedTask = completedTaskForProposal(0, proposal);
+    ExecutionTask inProgressTask = inProgressTaskForProposal(1, proposal2);
+
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 0, throttleRate);
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 1, throttleRate);
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 2, throttleRate);
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 3, throttleRate);
+    // Topic-level throttled replica config value should remain as "*"
+    assertExpectedThrottledReplicas(kafkaZkClient, TOPIC0, ReplicationThrottleHelper.WILD_CARD_ASTERISK);
+    assertExpectedThrottledReplicas(kafkaZkClient, TOPIC1, ReplicationThrottleHelper.WILD_CARD_ASTERISK);
+
+    throttleHelper.clearThrottles(Collections.singletonList(completedTask), Collections.singletonList(inProgressTask));
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 0, throttleRate);
+    // we expect broker 1 to be null since all replica movement related to it has completed.
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 1, null);
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 2, throttleRate);
+    // We expect broker 3 to have a throttle on it because there is an in-progress replica being moved
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 3, throttleRate);
+    // Topic-level throttled replica config value should remain as "*"
+    assertExpectedThrottledReplicas(kafkaZkClient, TOPIC0, ReplicationThrottleHelper.WILD_CARD_ASTERISK);
+    assertExpectedThrottledReplicas(kafkaZkClient, TOPIC1, ReplicationThrottleHelper.WILD_CARD_ASTERISK);
+
+    // passing an inProgress task that is not complete should have no effect.
+    throttleHelper.clearThrottles(Collections.singletonList(completedTask), Collections.singletonList(inProgressTask));
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 0, throttleRate);
+    // we expect broker 1 to be null since all replica movement related to it has completed.
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 1, null);
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 2, throttleRate);
+    // We expect broker 3 to have a throttle on it because there is an in-progress replica being moved
+    assertExpectedThrottledRateForBroker(kafkaZkClient, 3, throttleRate);
+    // Topic-level throttled replica config value should remain as "*"
+    assertExpectedThrottledReplicas(kafkaZkClient, TOPIC0, ReplicationThrottleHelper.WILD_CARD_ASTERISK);
+    assertExpectedThrottledReplicas(kafkaZkClient, TOPIC1, ReplicationThrottleHelper.WILD_CARD_ASTERISK);
+
+    // Completing the in-progress task and clearing the throttles should clean everything up.
```

Comment:
This comment is not correct right? -- i.e. it should _not_ clean everything as you have demonstrated below with the topic-level throttled replica config remaining as `WILD_CARD_ASTERISK`.

Reply:
Correct